draft-bormann-cbor-yang-standin.md

```diff
@@ -140,27 +141,98 @@ Tag 100 {{-date}} can unambiguously stand in for all `date-no-zone` values.
 
 YANG type | base type | specification | stand-in
 ip-address | union | {{-yang-types}} | (see union)
-ipv4-address | string | {{-yang-types}} | tag 52
 ipv6-address | string | {{-yang-types}} | tag 54
+ipv4-address | string | {{-yang-types}} | tag 52
 ip-address-no-zone | union | RFC 6991 | (see union)
-ipv4-address-no-zone | ipv4-address | RFC 6991 | tag 52
 ipv6-address-no-zone | ipv6-address | RFC 6991 | tag 54
+ipv4-address-no-zone | ipv4-address | RFC 6991 | tag 52
 ip-address-link-local | union | {{-legacy-bis}} | (see union)
-ipv4-address-link-local | ipv4-address | {{-legacy-bis}} | tag 52
 ipv6-address-link-local | ipv6-address | {{-legacy-bis}} | tag 54
+ipv4-address-link-local | ipv4-address | {{-legacy-bis}} | tag 52
 ip-prefix | union | {{-yang-types}} | (see union)
-ipv4-prefix | string | {{-yang-types}} | tag 52
 ipv6-prefix | string | {{-yang-types}} | tag 54
+ipv4-prefix | string | {{-yang-types}} | tag 52
 ip-address-and-prefix | union | {{-legacy-bis}} | (see union)
-ipv4-address-and-prefix | string | {{-legacy-bis}} | tag 52
 ipv6-address-and-prefix | string | {{-legacy-bis}} | tag 54
+ipv4-address-and-prefix | string | {{-legacy-bis}} | tag 52
 {: title="Legacy representations in ietf-yang-types"}
 
-TO DO: Define usage of tags 54 and 52 for the cases we actually
-support.
-Addresses that have zones given cannot use tag 54/52.
-Addresses with leading zeros cannot use tag 54/52.
-Addresses that do not comply with {{RFC5952}} cannot use tag 54.
+The encoder MAY normalize IPv6 addresses and prefixes that do not comply with {{RFC5952}}
+but can be converted into the stand-in representation. The encoder implementation SHOULD be clear
+about whether this normalization is employed or not.
+
+If the schema specifies `ip-prefix`, the encoder MAY normalize prefixes with
+non-zero bits after the prefix end. The encoder implementation SHOULD be clear
+about whether this normalization is employed or not.
+
+If the schema specifies a union between `ip-prefix` and `ip-address-and-prefix` (or a union of their subtypes),
+the encoder MUST prefer the `ip-address-and-prefix` stand-in over `ip-prefix`
+format. Specifically, it the schema allows to encode 2001:db8:1234::/48 both as
+`54([48, h'20010db81234'])` and `54([h'20010db81234', 48])`, the latter MUST be used.
+
+Adapted examples from {{-cbor-ip}}:
+
+Stand-in representation of IPv6 address 2001:db8:1234:deed:beef:cafe:face:feed
+is `54(h'20010db81234deedbeefcafefacefeed')`.
+
+CBOR encoding of stand-in (19 bytes):
+
+```
```

Comment:
The sourcecode-type for pretty-printed hex CBOR is "cbor-pretty"

So this should become ` ``` cbor-pretty ` 
(Don't know how to encode a suggestion with ` ``` ` in it...)

Reply:
fixed